go/vt/vtgate/endtoend/mirror_test.go

```diff
@@ -0,0 +1,180 @@
+package endtoend
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"math/rand"
+	osExec "os/exec"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"vitess.io/vitess/go/mysql"
+)
+
+var mirrorInitOnce sync.Once
+
+func BenchmarkMirror(b *testing.B) {
+	const numRows = 10000
+
+	ctx := context.Background()
+	conn, err := mysql.Connect(ctx, &vtParams)
+	if err != nil {
+		b.Fatal(err)
+	}
+	defer conn.Close()
+
+	var query bytes.Buffer
+
+	mirrorInitOnce.Do(func() {
+		b.Logf("seeding database for benchmark...")
+
+		for i := 0; i < numRows; i++ {
+			query.Reset()
+			query.WriteString(fmt.Sprintf("INSERT INTO ks.mirror_tbl1(id) VALUES(%d)", i))
+			_, err = conn.ExecuteFetch(query.String(), -1, false)
```

Comment:
I think we want tests to use `https://github.com/stretchr/testify` and write `require.NoError(t, err)` instead

Reply:
ah, it's a benchmark. ignore me